### PR TITLE
Enables hot reloading of mining configuration file

### PR DESCRIPTION
**Pull request by xoloki**
Enables hot reloading of mining configuration file.

Prior to every mining burnchain transaction, the config file will be reloaded if its modified time is newer and the following settings will be reloaded:

```
burnchain.burn_fee_cap
burnchain.satoshis_per_byte
burnchain.rbf_fee_increment
burnchain.max_rbf
node.microblock_frequency
node.wait_time_for_microblocks
```

The following config file items, if missing, will be read from environment variables:
```
node.seed => STACKS_NODE_SEED
burnchain.username => STACKS_BURNCHAIN_USERNAME
burnchain.password => STACKS_BURNCHAIN_PASSWORD 
```

Addresses the following issues:
- https://github.com/stacks-network/stacks-blockchain/issues/2629
- https://github.com/stacks-network/stacks-blockchain/issues/3181

### CODE REVIEW

**Comment by xoloki** [comment]
> Can you run stacks-node with a mock miner configuration to test the config reloading?

Sure, do you want me to write an integration test or just run it by hand and check?

File: testnet/stacks-node/src/config.rs [hunk 1]
```diff
@@ -363,6 +370,163 @@ pub struct Config {
...
+        let field2 = extractor(struct2);
+        if field1 != field2 {
+            info!("{} changed from {} to {}", field_name, field1, field2)
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ConfigLoaderHandle {
+    handle: std::sync::Arc<std::sync::Mutex<RefCell<ConfigLoader>>>,
```
Comment (jcnelson) [inline testnet/stacks-node/src/config.rs:494]: You probably don't need `RefCell` here.

**Comment by jcnelson** [comment]
On the whole, it's not clear to me why there needs to be a shared `ConfigHandle` structure.  The miner thread could simply load up the config file itself each time it gets spawned, which would obviate the need for any shared state in RAM.  It's not like it takes an extraordinary amount of time (especially when compared to mining).

File: testnet/stacks-node/src/config.rs (hunk 1 above)
Comment (xoloki) [reply testnet/stacks-node/src/config.rs]: RefCell removed from `ConfigHandle` and `ConfigLoaderHandle`.

**Comment by pavitthrap** [comment]
Closing for now, might resume progress in future.

**Pull request closed without merging**